### ISSUE TITLE
serviceAccount and main vault container name incompatible when using Consul Auto Inject 

### ISSUE THREAD

**Issue by lucasslima** [hashicorp/vault-helm#206]
Hi,

I'm making a POC around creating a Consul Connect mesh using Vault as a cert manager, and upon deploying Vault, the helm chart configures the serviceAccountName as
```
serviceAccountName: {{ template "vault.fullname" . }}
```
Which gets us something like "poc-vault", for instance. However, the name of the container is just "vault", and it was failing to register as a service on Consul since it's enforcing ACL rules with `deny` as default.

We've got around it using the annotation:
```
  # Extra annotations to attach to the server pods
  # This should be a multi-line string mapping directly to the a map of
  # the annotations to apply to the server pods
  annotations: |
    consul.hashicorp.com/connect-service: "{{ template "vault.fullname" . }}"
```
to the chart values. If the container name could be configured as the same name as the created service account would prevent this issue. 

In any event, hope this helps people doing something similar in the future. 

Cheers!

**Comment by malnick** [comment]
Thanks for this request @lucasslima - the take away here is that we want to enable this without modifying the existing template?

**Comment by lucasslima** [comment]
Yeah, I don't think you can get it to register on a ACL enabled Consul without making changes to the annotations as I mentioned.

**Comment by carrchang** [comment]
@lucasslima PR #367 resolved you requirement
